### PR TITLE
Add cross-platform path delimiter support

### LINKED ISSUE #14: Fix d2 location path resolution for cross-platform support
Presently the d2 Obsidian plugin seems to only function with *nix platforms due to the "env" path delimiter being hardcoded to ":"

This can be easily fixed by using Node's `path.delimiter` variable.

### PR DESCRIPTION
**Pull request by dlecklider**
Fixes #14

### CODE REVIEW

**Review by berniexie — APPROVED** [review]
💯

File: src/processor.ts [hunk 1]
```diff
@@ -123,13 +124,14 @@ export class D2Processor {
   };
 
   async generatePreview(source: string, signal: AbortSignal): Promise<string> {
+    const pathDelimiter = delimiter;
```
Comment (berniexie) [inline src/processor.ts:127]: Actually, why resassign to pathDelimiter? Why not just join on the import?

Comment (dlifty) [reply src/processor.ts:127]: @berniexie that may be a better way to do this. My development background isn't in Node/Web development so some of the more intimate details of the language/framework are not apparent to me. Want me to correct this, or you want to override yourself?

Comment (dlecklider) [reply src/processor.ts]: @berniexie I have made the requested changed. Cheers! 🍻

**Review by berniexie — APPROVED** [review]

**Pull request merged**